FR #966: Improving dataset metadata (#1464)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/appendix/metadata_examples/dataset_instance_metadata_example.md

# Dataset Instance Metadata

## Scenario

> ACME provides two different datset instances, "Cost and Usage Daily" and "Commitments," FOCUS Cost and Usage and FOCUS Commitments respectively.  ACME provides a directory of datasets metadat with a single file including each dataet instance's metadata.

Reviewer: ```suggestion
ACME provides two dataset instances, "Cost and Usage Daily" and "Commitments", each corresponding to the respective FOCUS datasets: Cost and Usage and Contract Commitment. ACME also provides a directory of datasets metadata containing a single file with metadata for each dataset instance.
```

## Supplied Metadata

Metadata can be provided at a location such as `/FOCUS/metadata/dataset_instances.json`.

The updated schema-related metadata could look like this:

```json
[
  {
    "DatasetInstanceId": "178151-dbad145e-178151-dbad145e-246811",

>     "DatasetInstanceName": "Contract Commitments Report",
>     "FOCUSDataset": "ContractCommitment"

Reviewer: ```suggestion
    "DatasetInstanceName": "Commitments",
    "DatasetId": "ContractCommitment"
```

  },

>   {

Author: Let add an hourly so they know more example of why we want it this way.

    "DatasetInstanceId": "178151-dbad145e-178151-dbad145e-178151",

>     "DatasetInstanceName": "COST_AND_USAGE_DAILY",

Reviewer: aligning with the name in the description

```suggestion
    "DatasetInstanceName": "Cost and Usage Daily",
```

Author: Trying to highlight they can name them whatever they want. leaving.

>     "FOCUSDataset": "CostAndUsage"

Reviewer: Consider using the term that we use in FOCUS dataset specifications

```suggestion
    "DatasetId": "CostAndUsage"
```

Author: I think this makes what it is less clear. We use this convention in other sections of the metadata...I am going to set this FOCUSDatasetId

# Dataset Instance Metadata

## Scenario

> ACME provides three dataset instances: "Cost and Usage Daily," "Cost and Usage Hourly," and "Contract Commitments," corresponding to the FOCUS datasets Cost and Usage and Contract Commitment. ACME also provides a metadata directory containing a single file with metadata for each dataset instance.

Reviewer 2: Now that we support multiple datasets, we should maybe define a style for when a FOCUS dataset is mentioned in a doc. I feel that datasets are at least as important as columns and glossary entries, and have their own definitions, we should treat them equally for better readability and understanding.

Apologies if this was discussed already elsewhere and I missed it.

Reviewer 3: Will address in a future consistency pass.  Resolving.

> ACME provides three dataset instances, "Cost and Usage Daily," "Cost and Usage Hourly," and "Commitments", each corresponding to the respective FOCUS datasets: Cost and Usage and Contract Commitment. ACME also provides a directory of datasets metadata containing a single file with metadata for each dataset instance.

Reviewer 4: ```suggestion
ACME provides three dataset instances: "Cost and Usage Daily," "Cost and Usage Hourly," and "Contract Commitments," corresponding to the FOCUS datasets Cost and Usage and Contract Commitment. ACME also provides a metadata directory containing a single file with metadata for each dataset instance.
```

## Supplied Metadata

Metadata can be provided at a location such as `/FOCUS/metadata/dataset_instances.json`.

The updated schema-related metadata could look like this:

```json
[
  {
    "DatasetInstanceId": "178151-dbad145e-178151-dbad145e-246811",
    "DatasetInstanceName": "Contract Commitments Report",

>     "FOCUSDatasetId": "ContractCommitment"

Reviewer: ```suggestion
    "FocusDatasetId": "ContractCommitment"
```

  },
  {
    "DatasetInstanceId": "178151-dbad145e-178151-dbad145e-178151",
    "DatasetInstanceName": "Cost and Usage Daily",

>     "FOCUSDatasetId": "CostAndUsage"

Reviewer: ```suggestion
    "FocusDatasetId": "CostAndUsage"
```

  },
  {
    "DatasetInstanceId": "178151-ja23h1287-387151-dbad145e-134657",
    "DatasetInstanceName": "Cost and Usage Hourly",

>     "FOCUSDatasetId": "CostAndUsage"

Reviewer: ```suggestion
    "FocusDatasetId": "CostAndUsage"
```
